### DISCUSSION TITLE
Linking tables from queries

### DISCUSSION THREAD

**Discussion by mpolak83** [Budibase/budibase#8638] [Help] [answered]
Hi guys,

I'm new to budibase and I'd appreciate your help on the following:
I have connected to PostgreSQL db and created query that pulls all sales orders headers from database SOH table. Headers provide info like sales order number, who loaded it etc.
SOL table contains sales lines that are related to its SOH. So SOH there will be one entry, but in SOL there will be 1 one more lines that link to  SOH by record number.

I want to have a View button in each header line, and when I click on button it will open modal table with relevant lines for that SOH.

I have created SQL queries to get data from both tables. I cannot use native tables as there are tens of relations and 80% data I do not need to see. 
Query to display SOH:

`SELECT`
`soh._ordernum AS "Sales order number",`
`csh._name AS "Customer"`
`FROM data.soh`
`LEFT JOIN data.csh ON data.soh._customer = data.csh._recno`
Query to display all SOLs for given SOH:

`SELECT `
	`data.sol._orderref AS "Sorder",`
 	`data.sol._line AS "Line number",`
	`data.mch._material AS "Our Part number"`
	
`FROM data.sol`
`LEFT JOIN data.mch ON data.sol._ourpartnum = data.mch._recno`
`WHERE data.sol._orderref = {{sol_by_soh}}`

Any ideas how to accomplish that?

Thank you

Marcin

**Comment by melohagan** [comment, marked as answer]
Hey Marcin, 

You could have a view screen that lists all of your sales orders headers, via a data provider using the sales order headers query as it's source, with a table nested underneath.

Next you would add a button to each row of your table. You do that by clicking on the table component, and then adding the button component in the side panel.
<img width="198" alt="Screenshot 2022-11-21 at 09 29 05" src="https://user-images.githubusercontent.com/101575380/203014693-2cece4d2-0180-4173-a284-dd99dc3ce270.png">

**Define actions** for your button like so:
<img width="914" alt="Screenshot 2022-11-21 at 09 30 18" src="https://user-images.githubusercontent.com/101575380/203014941-e46728dd-1671-4152-a5f1-338f156cdc91.png">

Then in your `/sol/:orderref` screen you would use your custom query and pass the URL var as your `sol_by_soh` binding.
<img width="914" alt="Screenshot 2022-11-21 at 09 38 40" src="https://user-images.githubusercontent.com/101575380/203016884-4baa709b-c8c0-4806-b491-41c1015d73d1.png">

A bit more on custom queries here: https://docs.budibase.com/docs/custom-queries

Make sure you have set the route correctly in your order details page:
<img width="260" alt="Screenshot 2022-11-21 at 09 43 21" src="https://user-images.githubusercontent.com/101575380/203017939-b89419e4-aa60-47fb-a0c5-04f474c131df.png">

**Reply by mpolak83** [reply]
Thank @melohagan you for your help. That worked perfectly.
